File: azurefunction1053/travelcardservice/Models/CardholderDto.cs
namespace travelcardservice.Models
{
    public class CardholderDto
    {
        public string CardholderTitle { get; set; } = string.Empty;
        public string CardholderForename { get; set; } = string.Empty;
        public string CardholderSurname { get; set; } = string.Empty;
        public CardholderType CardholderType { get; set; } = CardholderType.Primary;
        public string CardholderPhotoName { get; set; } = string.Empty;
        public string CardholderPhotoRRSKey { get; set; } = string.Empty;
        public string CardholderPhotoURL { get; set; } = string.Empty;
        public string CardholderPhotoKey { get; set; } = string.Empty;
    }
}

File: src/main/java/com/ai2dev/test_sb_codex/dto/CardholderDto.java
package com.ai2dev.test_sb_codex.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public record CardholderDto(
    @NotBlank
    @Size(min = 1, max = 15)
    String cardholderTitle,

    @NotBlank
    @Size(min = 1, max = 100)
    String cardholderForename,

    @NotBlank
    @Size(min = 1, max = 100)
    String cardholderSurname,

    @NotBlank
    @Pattern(regexp = "^(Primary|Secondary)$")
    String cardholderType,

    @NotBlank
    @Size(min = 1, max = 100)
    String cardholderPhotoName,

    @Size(min = 39, max = 42)
    String cardholderPhotoRRSKey,

    @Size(min = 20, max = 2048)
    @Pattern(regexp = "^(https?://).*")
    String cardholderPhotoURL,

    @Size(min = 39, max = 42)
    String cardholderPhotoKey
) {}
